flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/reader/MySqlRecordEmitter.java

```diff
@@ -120,7 +123,31 @@ private void emitElement(SourceRecord element, SourceOutput<T> output) throws Ex
         debeziumDeserializationSchema.deserialize(element, outputCollector);
     }
 
-    private void reportMetrics(SourceRecord element) {
+    private void reportMetrics(SourceRecord element, MySqlSplitState splitState) {
+        TableId tableId = RecordUtils.getTableId(element);
+        if (splitState.isSnapshotSplitState()) {
+            sourceReaderMetrics.numRecordsOutSnapshotIncrease(tableId);
+        } else if (splitState.isBinlogSplitState()) {
+            if (RecordUtils.isDataChangeRecord(element)) {
+                Struct value = (Struct) element.value();
+                if (value != null) {
+                    Envelope.Operation operation =
+                            Envelope.Operation.forCode(
+                                    value.getString(Envelope.FieldName.OPERATION));
+                    switch (operation) {
+                        case CREATE:
+                            sourceReaderMetrics.numRecordsOutInsertIncrease(tableId);
+                            break;
+                        case UPDATE:
+                            sourceReaderMetrics.numRecordsOutUpdateIncrease(tableId);
+                            break;
+                        case DELETE:
+                            sourceReaderMetrics.numRecordsOutDeleteIncrease(tableId);
+                            break;
+                    }
+                }
+            }
```

Comment:
We checked whether element is data change record or not in line 101.
How about removing this condition?

```suggestion
            Struct value = (Struct) element.value();
            if (value != null) {
                Envelope.Operation operation =
                        Envelope.Operation.forCode(
                                value.getString(Envelope.FieldName.OPERATION));
                switch (operation) {
                    case CREATE:
                        sourceReaderMetrics.numRecordsOutInsertIncrease(tableId);
                        break;
                    case UPDATE:
                        sourceReaderMetrics.numRecordsOutUpdateIncrease(tableId);
                        break;
                    case DELETE:
                        sourceReaderMetrics.numRecordsOutDeleteIncrease(tableId);
                        break;
                }
            }
```

Reply:
Thank you very much for helping me review the code.
I will delete the isDataChangeRecord judgment